### DISCUSSION TITLE
In album list view how to sort correctly first by original date and then by date

### DISCUSSION THREAD

**Discussion by Groxxxy** [TT-ReBORN/Georgia-ReBORN#39] [Q&A] [answered]
On the right of the screen in the list view the year of multiple albums is not placing them in chronological order.

How is it possible to allow the albums in the list to sort chronologically first by the existence of an %originaldate% tag and then by %date% ?

The intention is because not all albums have %originaldate%, only those that are rereleases have both originaldate and date, it is preferred to sort by originaldate if it exists and factor that into album sort.

Interestingly this is already implemented on the details tab, if %originaldate% exists it is preferred and therefore shown instead of %date%.

See line 47 in: georgia-reborn-config.jsonc

`{ "label": "Year", "val": "$puts(d,$if3(%original release date%,%originaldate%,%date%,%fy_upload_date%,))$if($strcmp($year($get(d)),$get(d)),$get(d),)" }, // 'Year' is shown if the date format is YYYY`

This problem is actually seen in the official screenshots on the right (e.g. 2015, 2014, 2011, 2016):
https://camo.githubusercontent.com/5439e05ce6c5e436ce6aff8ca9c21c21c9825cfa8d7c73c23b3ba67cf73d53f8/68747470733a2f2f692e6962622e636f2f77364c71727a502f47656f7267652d52652d424f524e2d5468656d65732d416e696d6174696f6e2e77656270

![image](https://user-images.githubusercontent.com/14025275/146032915-36e04820-c59f-429d-8b93-0c489d35e5c7.png)
(e.g. 1973, 1987, 1972)

**Comment by TT-ReBORN** [comment]
Hi @Groxxxy,

you can do that by right click in the playlist for context menu and select:
Sort > by date
Sort > by... ( for custom sorting pattern ).

And if you want you can reverse the date order:
Sort > Reverse

I didn't do that in the screenshots, nor in my playlists, because I prefer my playlist order manually arranged...

-TT

**Comment by Groxxxy** [comment]
Hi TT, thank you for responding. I have %date% vorbis comment for all albums and only one %originaldate% for a 1989 album because it was rereleased in 1998.

A basic sort by date gets the order confused and places the 1989 album after 1998.
![l1ywf17jzU](https://user-images.githubusercontent.com/14025275/146109833-646e7f7b-904a-4b8a-9925-ce1ebf27caeb.gif)

Sort > by %ORIGINALDATE% - %DATE% - %ALBUMARTIST% - %ALBUM% - %DISCNUMBER% - %TRACKNUMBER% - %TITLE%

Places 1989 at the very end after an album with %date% 2011:
![jiqtiCgUIb](https://user-images.githubusercontent.com/14025275/146110198-739081ad-1ac4-4a6c-8513-9d034798476c.gif)

For this to be correct Sort > by %ORIGINALDATE% - %DATE% needs to be respected and Sort > by date needs to... sort by date.

Thank you!

**Reply by TT-ReBORN** [reply]
Actually there is nothing wrong in the animations you've shown here.
It does sort by %date% in your first animation ( %date% = 1998 ) but the %originaldate% will be always displayed first
and it does sort by %originaldate% in your second animation ( after all tagged %date% tags, you can reverse the order though ).
That's because it is defined in the georgia-reborn-config.jsonc.

%originaldate% or %original release date% is the right tagging formula you should use.
If you don't want to retag your files you can use this not recommended method, open your georgia-reborn-config.jsonc:

In the "settings": part replace it with this:
`"defaultSortString": "$if2(%artist sort order%,%album artist%) $if3(%album sort order%,%date%,%original release date%) %album% %edition% %codec% %discnumber% %tracknumber%",`

and in "title_format_strings":
`"date": "$if3(%date%,%original release date%,%originaldate%,%fy_upload_date%,)",`

Now sort in the playlist by date.

@kbuffington Is it even possible to merge %originaldate% with %date% in the sorting pattern?

**Comment by Groxxxy** [comment]
Yes, I understand what you are saying. I do actually use %originaldate%.

My recommendation is that sort follows originaldate then date. I believe in "title_format_strings" "date" is only a display, not a sorting.
![image](https://user-images.githubusercontent.com/14025275/146721089-30cc930a-2583-4661-a11b-e4311eafc77f.png)
(Originaldate is 1989 but date for this album is 1998, it is displaying 1989 but sorting by 1998)

I have read titleformat_help.html. I believe if I use $if(X,Y,Z) and:

`"date": "$if3(%originaldate%,%originaldate%,%date%,)",`

Sort should sort by originaldate first otherwise by date second. But where do I actually change the sorting instead of simply display via "title_format_strings"?

**Reply by TT-ReBORN** [reply]
It does work how you want, first check your settings in your georgia-reborn-config.jsonc.
The default settings should look like this:

In the "settings":
`"defaultSortString": "$if2(%artist sort order%,%album artist%) $if3(%album sort order%,%original release date%,%date%) %album% %edition% %codec% %discnumber% %tracknumber%",`

and in "title_format_strings":
`"date": "$if3(%original release date%,%originaldate%,%date%,%fy_upload_date%,)",`

Next in the playlist right click > Sort > by...
`%ORIGINALDATE% - %DATE% - %ALBUM% - %DISCNUMBER% - %TRACKNUMBER% - %TITLE%`

**Then right click again Sort > Reverse**


And it should look like this:


Sort by Original Date, Date ( Artist: Envy, Selenites with Original Date, Dreamwell, Thou without Original Date,  ):
![Sort1](https://user-images.githubusercontent.com/79368795/146771457-c88f4674-47fe-45d2-9a6c-aebf1c4feaca.jpg)


Sort by Original Date, Date ( Artist: Dreamwell with Original Date and Date, Envy, Selenites with Original Date, Thou with Date ):
![Sort2](https://user-images.githubusercontent.com/79368795/146771593-d501de58-bc04-4b86-921c-6d0bd012c267.jpg)


As you can see it does sort by Original Date first and then Date.

I guess foobar's default sorting order is confusing people because it is reversed...

-TT

**Comment by Groxxxy** [comment]
**These are your albums:** 

Dreamwell - Modern Grotesque (2021)
`<ORGINALDATE> 2021`

Selenites - Jeder für sich und Gott gegen alle (2003)
`<ORGINALDATE> 2001`

Envy - A Dead Sinking Story (2002) // Error here, how are you personally using %date%; %originaldate% should be older than %date% but yours here is younger.
`<ORGINALDATE> 2003`

**Sorted and displayed by original date and date correctly it would look like this (oldest first):**

Selenites - Jeder für sich und Gott gegen alle (2003)
**Sorted and displaying:** `<ORGINALDATE> 2001`

Envy - A Dead Sinking Story (2003) 
**Sorted and displaying:** `<ORGINALDATE> 2002` // _Corrected_

Dreamwell - Modern Grotesque (2021)
**Sorted and displaying:** `<ORGINALDATE> 2021`

**That's your first issue. Secondly I only use originaldate when the album has been rereleased at a new year. Typically I only use the date tag but when I use originaldate in combination with date I need it to sort by originaldate, essentially a good sort will check for the existence of originaldate and sort by it instead of date.**

I really thought this would be possible through "defaultSortString" and "title_format_strings" via but nothing changes with:

`"date": "$if3(%originaldate%,%originaldate%,%date%,)",`

![146771457-c88f4674-47fe-45d2-9a6c-aebf1c4feaca](https://user-images.githubusercontent.com/14025275/147868641-ad7fa8d1-17bd-4a4e-a2ba-f752e8deca37.png)

Excuse me, but is this chronological? Reverse or not reversed that's not what I mean. The displayed year in this situation cannot be chronologically ordered. How do you make it be chronologically sorted and displayed in this particular situation?

**Comment by TT-ReBORN** [comment, marked as answer]
Sorry to confused you about the example above, I thought that my example was clear enough...
I have put intentionally the wrong year in the Envy album to demonstrate that it will sort first by %originaldate% then %date% which it actually does. In the example here ( the one you marked with ??? ) the first 2 albums from the top are tagged with %originaldate% then followed by the other two by %date%. And yes the year is chronologically ordered, by default it's reversed ( oldest first, latest last ).

But never mind all that for now, I have actually found a way to sort it as you want.
If %originaldate% exist it will prioritize this tag first then the %date% tag and sort accordingly within %date% tagged albums.
You just need to put the same syntax in the playlist sort pattern, right click in playlist > Sort > by...
`"$if3(%original release date%, %originaldate%, %date%) %album% %edition% %codec% %discnumber% %tracknumber%"`
copy the whole syntax with both "".

I'll make this sort pattern default for playlist > Sort > by date.

Edit: It will now work when you sort by date in the playlist. Just download the latest version from [here](https://github.com/TT-ReBORN/Georgia-ReBORN/archive/refs/heads/master.zip) and replace your existing files.

-TT

**Comment by Groxxxy** [comment]
I did not know you can put that syntax into the playlist sort pattern 👍 
Perfect player now, thanks!
Happy new year TT!

**Reply by TT-ReBORN** [reply]
Thanks and Happy New Year to you too =)

**Comment by TT-ReBORN** [comment]
Btw I forgot to tell you:

You tag with %original release date% = when first released and %date% = remaster year and %edition% = Deluxe edition or Special Edition. If you do it this way, you have it really nice displayed in the Playlist and in Details.

**Comment by Groxxxy** [comment]
Hi TT, I'm back again.

How do I make Sort > Date stick permanently for all playlists because it resets currently?

Thanks.

![foobar2000_piU2z65nfC](https://user-images.githubusercontent.com/14025275/183250165-f7677bf3-fe4b-4f85-ad0f-901b3156b27c.gif)

**Reply by TT-ReBORN** [reply]
Hi @Groxxxy,

thanks for pointing it out, you're right about that. I'll make it default behavior when adding items from the Library to the Playlist.
This happens because albums are not automatically sorted by date in the Library. This will be available in the first public release, 
i.e next update.

-TT

**Comment by TT-ReBORN** [comment]
Hey @Groxxxy,

I've added an automated sort order ( 'Default', 'Date ascending', 'Date descending', 'Date displayed ascending', 'Date displayed descending' ) for the Library, can be set in top menu Options > Library > Sort order or in the Library hamburger menu.
With this set you can achieve exactly what you want.

There is also a manual playlist sort order in top menu Options > Playlist > Sort order.

If you want you can download and test it out, check the BETA thread [here](https://github.com/TT-ReBORN/Georgia-ReBORN/discussions/74).

-TT

**Comment by Groxxxy** [comment]
### ### I am using file structure view.

"Library, can be set in top menu Options > Library > Sort order" > Basically it sorts depending on what it sees in file structure view and sort order doesn't change that when it translates to the playlist if I am to play the entire discography.

![image](https://user-images.githubusercontent.com/14025275/190851079-be7361d7-010e-409e-94f2-8eab602cc3cb.png)

![foobar2000_MuM0sQmEvp](https://user-images.githubusercontent.com/14025275/190851030-25346b54-af17-4063-b565-5f82d86095bf.gif)

Yes, it will sort the playlist when it has been added but this requires manual work.
![foobar2000_K7lAYZYkXZ](https://user-images.githubusercontent.com/14025275/190851210-2546dcca-1279-4b58-b8a4-79df21786cd9.gif)

How to make it sort by original date in the file structure view so it goes to playlist in the correct sort without manual intervention?

### ### View by Album Artist |  Album > Options > Library > Sort order > Ascending (show year)
Pauls's Boutique was released after Licensed to Ill originally, my Paul's Boutique is a reissue from 1998 - again it would help to have _Options > Library > Sort order_ sort by original date first if it exists then by date.
![image](https://user-images.githubusercontent.com/14025275/190852323-d81a1b53-f152-49dc-84e0-5b0520442a22.png)

However primarily I am interested in a file structure sort in library view, then again I don't know how other people view their libraries. I have always organised albums into artist folders then viewed folder structure - now just to have those albums sort into playlist view by original date.

**Reply by TT-ReBORN** [reply]
Hi @Groxxxy,

Library's "View by Folder Structure" is not configurable with any sorting, it would defeat it's purpose...

The standard folder structure for any artist on your hard disk should be the following:

Beastie Boys
|_ 1986 - Licensed to III
|_ 1989 - Paul's Boutique ( 1998 Reissue ) // *
|_ 1992 - Check Your Head
|_ 1994 - III Communication
|_ 1998 - Hello Nasty
|_ 2004 - To the 5 Boroughs
|_ 2007 - The Mix-Up
|_ 2011 - Hot Sauce Committee Part Two

// * You do the same for remaster editions and for tagging 1989 = %original release date% or %originaldate% and 1998 for %date%.

If you have this folder structure, it will also show it correctly when using the Library's folder structure view and then add it 
to the Playlist.

The other Library views should indeed sort first by %original release date% or %originaldate% , 
@Wil-B, we should use this in the sorting instead of just using %date%:
`'$if3(%original release date%,%originaldate%,%date%,%fy_upload_date%,)'`

I'll add it in the next update.

-TT
